var/spack/repos/builtin/packages/py-globus-sdk/package.py

```diff
@@ -32,3 +33,4 @@ class PyGlobusSdk(PythonPackage):
     depends_on("py-pyjwt@2.0.0:2+crypto", type=("build", "run"))
     depends_on("py-cryptography@3.3.1:3.3,3.4.1:", when="@3.7:", type=("build", "run"))
     depends_on("py-cryptography@2:3.3,3.4.1:3.6", when="@:3.0", type=("build", "run"))
+    depends_on("py-typing-extensions@4:", when="@3.25:", type=("build", "run"))
```

Comment:
Some of the `py-globus-sdk` dependencies need to be conditioned for the new version based on `setup.cfg`:

```
[options]
python_requires = >=3.7
package_dir = 
    =src
packages = find:
include_package_data = true
install_requires = 
    requests>=2.19.1,<3.0.0
    pyjwt[crypto]>=2.0.0,<3.0.0
    cryptography>=3.3.1,!=3.4.0
    typing_extensions>=4.0; python_version<"3.10"
```